pkg/mcoproducer/mcoproducer.go

```diff
@@ -0,0 +1,146 @@
+package mcoproducer
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/utils"
+)
+
+const replaceKernelModuleScriptTemplate = `#!/bin/bash
+echo "before checking podman images"
+if podman images | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry, removing in-tree kernel module"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -rd /opt $KERNEL_MODULE
+    if [ $? -eq 0 ]; then
+            echo "Succesffully removed the in-tree kernel module $KERNEL_MODULE"
+    else
+            echo "failed to remove in-tree kernel module $KERNEL_MODULE"
+    fi
+    echo "Running container image to insert the oot kernel module $KERNEL_MODULE"
+    podman run --privileged --entrypoint modprobe $IMAGE:$TAG -d /opt $KERNEL_MODULE
+    echo "OOT kernel module $KERNEL_MODULE is inserted"
+else
+   echo "Image $IMAGE:$TAG is not present in local registry, will try after reboot"
+fi
+`
+const pullKernelModuleScriptTemplate = `#!/bin/bash
+if podman image list | grep $IMAGE | grep -q $TAG; then
+    echo "Image $IMAGE:$TAG found in the local registry.Nothing to do"
+else
+    echo "Image $IMAGE:$TAG not found in the local registry, pulling"
+    podman pull --authfile /var/lib/kubelet/config.json $IMAGE:$TAG
+    if [ $? -eq 0 ]; then
+        echo "Image $IMAGE:$TAG has been successfully pulled, rebooting.."
+        reboot
+    else
+        echo "Failed to pull image $IMAGE:$TAG"
+    fi
+fi
+`
+
+const mcoTemplate = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: $MCO_MACHINE_CONFIG_POOL_REF
+  name: $MCO_NAME
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+        - contents: |
+            [Unit]
+            Description=Replace in-tree kernel module with oot kernel module
+            Before=network-pre.target
+            Wants=network-pre.target
+            DefaultDependencies=no
+            [Service]
+            User=root
+            Type=oneshot
+            TimeoutSec=10
+            ExecStartPre=ls /usr/local/bin
```

Comment:
Why do we need it?

Reply:
mainly for debugging: it does not cost much, but it will allow us to see the contents of the /usr/local/bin directory in case something is not working